app/models/miq_expression.rb

```diff
@@ -1593,6 +1593,31 @@ def self._quick_search?(e)
     end
   end
 
+  def fields
+    self.class._fields(exp)
+  end
+
+  def self._fields(expression)
+    result = []
+    case expression
+    when Array
+      expression.each { |x| result.concat(_fields(x)) }
+    when Hash
+      if expression.keys.include?('field')
+        result.push(expression['field'])
+      else
+        result.concat(_fields(expression.values))
```

Comment:
I think you'll need to be explicit in asking if this is an `AND`, `OR` or `!` (including aliases of those) before recurring - absense of the `field` key may not be sufficient